icu4c/source/i18n/numsys.cpp

```diff
@@ -202,10 +209,17 @@ NumberingSystem::createInstanceByName(const char *name, UErrorCode& status) {
 
     if (U_FAILURE(status)) {
         status = U_UNSUPPORTED_ERROR;
```

Comment:
Assigning status here, when it's already showing another error, is a little iffy. It could be overwriting a U_MEMORY_ALLOCATION_ERROR, for example. On the other hand, turning some sort of resource loading error into a U_UNSUPPORTED_ERROR was probably intended. Might want to get Shane's (sffc) opinion.

Reply:
Thanks Andy! I wonder if perhaps specifically checking for U_MEMORY_ALLOCATION_ERROR, and passing that back to the caller might be better/worthwhile? It does seem like it might be worth reporting back such a catastrophic failure to the caller. (Adding in @sffc to see what he might think as well).